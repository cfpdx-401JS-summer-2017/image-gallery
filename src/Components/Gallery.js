import React, { Component } from 'react';
import PropTypes from 'prop-types';

GalleryItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    bunny: PropTypes.object.isRequired,
    onRemove: PropTypes.func.isRequired
}

export function GalleryItem({ title, description, url, bunny, onRemove, onUpdate, bunnyNum }) {
    return (
        <div className="galleryView">
            <img src={url} alt={description} />
            <h3>{title}</h3>
            <p>{description}</p>
            <button onClick={() => onRemove(bunny)} >X</button>
        </div>  
    );
}

export class Gallery extends Component {
    
    static propTypes = {
        galleryArray: PropTypes.arrayOf(PropTypes.object).isRequired,
        onRemove: PropTypes.func.isRequired
    }
    
    render() {
        const { galleryArray, onRemove, onUpdate, bunnyNum } = this.props;
        const bunnyGallery = galleryArray.map((bunny, i) => (
            <GalleryItem key={i} bunny={bunny} bunnyNum={bunnyNum} title={bunny.title} description={bunny.description} url={bunny.url} onRemove={onRemove} onUpdate={onUpdate} />
        ));
        return (
            <div>
                {bunnyGallery[bunnyNum]}
                <button onClick={() => onUpdate(bunnyNum - 1)}>Previous Bunny</button>
                <button onClick={() => onUpdate(bunnyNum + 1)}>Next Bunny</button>
            </div>
        );
    }
}