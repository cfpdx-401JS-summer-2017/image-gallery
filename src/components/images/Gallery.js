import React, { Component } from 'react';
import PropTypes from 'prop-types';

GalleryItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    bunny: PropTypes.object.isRequired,
    onRemove: PropTypes.func.isRequired
}

export function GalleryItem({ title, description, url, bunny, onRemove, onUpdate }) {
    return (
        <div className="galleryView">
            <img src={url} alt={description} />
            <h3>{title}</h3>
            <p>{description}</p>
            <button onClick={() => onRemove(bunny)} >X</button>
        </div>  
    );
}

export class Gallery extends Component {
    
    static propTypes = {
        galleryArray: PropTypes.arrayOf(PropTypes.object).isRequired,
        onRemove: PropTypes.func.isRequired
    }
    
    render() {
        const { galleryArray, onRemove, onUpdate, bunnyNum } = this.props;
        const bunnyGallery = galleryArray.map((bunny, i) => (
            <GalleryItem key={i} bunny={bunny} title={bunny.title} description={bunny.description} url={bunny.url} onRemove={onRemove} onUpdate={onUpdate} />
        ));
        return (
            <div>
                <nav id="galleryNav">
                    <button onClick={() => onUpdate(-1)}>&laquo; Previous</button> <button onClick={() => onUpdate(1)}>Next &raquo;</button>
                    <p id="bunnyCounter">Bunny {bunnyNum + 1} of {bunnyGallery.length}</p>
                </nav>
                {bunnyGallery[bunnyNum]}
            </div>
        );
    }
}