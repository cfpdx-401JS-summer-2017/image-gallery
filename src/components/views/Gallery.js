import React from 'react';
import PropTypes from 'prop-types';

Gallery.propTypes = {
  deleteImage: PropTypes.func,
  images: PropTypes.array
};

export default function Gallery({ deleteImage, images }) {

  return (
    <div className="galleryView">
      {images.map((image, i) => (
        <div key={i}>
          <input type="checkbox" onChange={() => deleteImage(image)} />
          <img
            value={i}
            className="galleryImg"
            alt={image.alt}
            src={image.url}
          />
        </div>
      ))}
    </div>
  );
}
