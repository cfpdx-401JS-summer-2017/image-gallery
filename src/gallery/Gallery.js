import React, { Component } from 'react';
import PropTypes from 'prop-types';


export default class Gallery extends Component {

    static propTypes = {
        bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }

    render() {
        const bunny = this.props.bunnies[this.props.current];

        return (
            <div>
                <img src={bunny.url} alt={bunny.title}/>
                <h5>{bunny.title}</h5>
                <button 
                    value="Previous"   
                    onClick={(event) => this.props.cycleView(event.target.value)}>
                    Previous
                </button>
                <button 
                    value="Next" 
                    onClick={(event) => this.props.cycleView(event.target.value)}>
                    Next
                </button>
                <br></br>
                <button 
                    onClick={(event) => this.props.onRemove(bunny._id)} >
                    Remove this Bunny
                </button>
            </div>
        )
    }
}