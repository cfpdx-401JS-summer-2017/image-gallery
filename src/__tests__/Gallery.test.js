import React from 'react';

import { shallow } from 'enzyme';

import toJSON from 'enzyme-to-json';
import Bunnies from '../images/bunnies';
import Gallery from '../gallery/Gallery';

describe( 'Gallery', () => {

    it( 'displays bunnies in a gallery', () => {
        const wrapper = shallow(<Gallery bunnies={Bunnies} current={0}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});