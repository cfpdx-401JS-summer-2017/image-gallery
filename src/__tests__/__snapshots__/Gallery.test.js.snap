// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem tests displays correct properties 1`] = `
<div
  className="galleryView"
>
  <img
    alt="Test description"
    src="http://google.com/"
  />
  <h3>
    Test Title
  </h3>
  <p>
    Test description
  </p>
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
